File: 01-Arrays/20250729-beginner/go/solution_test.go
package main

import "testing"

func TestArrayContains(t *testing.T) {
	tests := []struct {
		name     string
		arr      []int
		target   int
		expected bool
	}{
		{
			name:     "Empty array",
			arr:      []int{},
			target:   5,
			expected: false,
		},
		{
			name:     "Single element - found",
			arr:      []int{42},
			target:   42,
			expected: true,
		},
		{
			name:     "Single element - not found",
			arr:      []int{42},
			target:   10,
			expected: false,
		},
		{
			name:     "Multiple elements - target at beginning",
			arr:      []int{1, 2, 3, 4, 5},
			target:   1,
			expected: true,
		},
		{
			name:     "Multiple elements - target at end",
			arr:      []int{1, 2, 3, 4, 5},
			target:   5,
			expected: true,
		},
		{
			name:     "Multiple elements - target in middle",
			arr:      []int{1, 2, 3, 4, 5},
			target:   3,
			expected: true,
		},
		{
			name:     "Multiple elements - target not found",
			arr:      []int{1, 2, 3, 4, 5},
			target:   10,
			expected: false,
		},
		{
			name:     "Array with negative numbers - found",
			arr:      []int{-5, -2, -8, -1},
			target:   -2,
			expected: true,
		},
		{
			name:     "Array with negative numbers - not found",
			arr:      []int{-5, -2, -8, -1},
			target:   2,
			expected: false,
		},
		{
			name:     "Array with duplicates - found",
			arr:      []int{7, 7, 7, 7},
			target:   7,
			expected: true,
		},
	}

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			result := ArrayContains(tt.arr, tt.target)
			if result != tt.expected {
				t.Errorf("ArrayContains(%v, %d) = %t, expected %t", tt.arr, tt.target, result, tt.expected)
			}
		})
	}
}

File: 01-Arrays/20250729-beginner/ts/solution.test.ts
import { arrayContains } from './solution';

describe('arrayContains', () => {
    test('Empty array', () => {
        expect(arrayContains([], 5)).toBe(false);
    });

    test('Single element - found', () => {
        expect(arrayContains([42], 42)).toBe(true);
    });

    test('Single element - not found', () => {
        expect(arrayContains([42], 10)).toBe(false);
    });

    test('Multiple elements - target at beginning', () => {
        expect(arrayContains([1, 2, 3, 4, 5], 1)).toBe(true);
    });

    test('Multiple elements - target at end', () => {
        expect(arrayContains([1, 2, 3, 4, 5], 5)).toBe(true);
    });

    test('Multiple elements - target in middle', () => {
        expect(arrayContains([1, 2, 3, 4, 5], 3)).toBe(true);
    });

    test('Multiple elements - target not found', () => {
        expect(arrayContains([1, 2, 3, 4, 5], 10)).toBe(false);
    });

    test('Array with negative numbers - found', () => {
        expect(arrayContains([-5, -2, -8, -1], -2)).toBe(true);
    });

    test('Array with negative numbers - not found', () => {
        expect(arrayContains([-5, -2, -8, -1], 2)).toBe(false);
    });

    test('Array with duplicates - found', () => {
        expect(arrayContains([7, 7, 7, 7], 7)).toBe(true);
    });
});